tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/ToEtcExpr.scala

```diff
@@ -466,9 +466,11 @@ class ToEtcExpr(annotationStore: AnnotationStore, aliasSubstitution: ConstSubsti
         val a = varPool.fresh
         val b = varPool.fresh
         val c = varPool.fresh
+        // The possible types to which which DOMAIN can be be applied,
+        // and the corresponding type of the domain when so applied:
         val funType = OperT1(Seq(FunT1(a, b)), SetT1(a)) // (a -> b) => Set(a)
         val seqType =
-          OperT1(Seq(SeqT1(a)), SetT1(IntT1())) // Seq(c) => Set(Int)
+          OperT1(Seq(SeqT1(c)), SetT1(IntT1())) // Seq(c) => Set(Int)
         val recType = OperT1(Seq(RecT1()), SetT1(StrT1())) // [] => Set(Str)
```

Comment:
This is actually a small logic fix, that was caught by the unused removal trying to eliminate the `c`. I don't think it could have caused problems in our type checking, since only one of *either* the `funType` or the `seqType` could be relevant for any application of the domain operator, and `b` is always fresh in any case, but nice to have it cleaned up just in case.

Reply:
There’s a failing unit test due to this change. I’ll fix it tomorrow.

Reply:
Yes, it was supposed to be a fix, but the actual variable was not substituted.